### DISCUSSION TITLE
how to get correct version of frontend?

### DISCUSSION THREAD

**Discussion by j-carson** [jupyterlab/jupyterlab-git#1166] [Q&A]
Did the install process change?

I installed using `pip install jupyterlab-git`

Starting up I get the error: The versions of the JupyterLab Git server frontend and backend do not match. The @jupyterlab/git frontend extension has version: 0.38.0 while the python package has version 0.39.2. Please install identical version of jupyterlab-git Python package and the @jupyterlab/git extension. Try running: pip install --upgrade jupyterlab-git

I can pip install jupyterlab-git=0.38.0 and that works, but I can't figure out the correct command to get the frontend to 0.39.2

**Comment by fcollonval** [comment]
@j-carson could you try installing from scratch in a virtual environment `pip install jupyterlab jupyterlab-git`? If that works, you can force uninstalling jupyterlab-git with `pip uninstall jupyterlab-git`. Then you can check if `jupyter labextension list` does not report it. If it does, please report here what you are seeing.

**Reply by j-carson** [reply]
Oh... weird.
`pip list | grep jupyterlab` shows 
```
jupyterlab             3.4.7
jupyterlab-git         0.39.2
jupyterlab-pygments    0.2.2
jupyterlab_server      2.15.1
```
`jupyter labextension list` shows
```
JupyterLab v3.4.7
/home/xxx/janet/.local/share/jupyter/labextensions
        jupyterlab_pygments v0.2.2 enabled OK (python, jupyterlab_pygments)
        nbdime-jupyterlab v2.1.1 enabled OK
        @jupyterlab/git v0.38.0 enabled OK (python, jupyterlab-git)
        @axlair/jupyterlab_vim v0.15.1 enabled OK (python, jupyterlab_vim)

Other labextensions (built into JupyterLab)
   app dir: /home/echodyne.int/janet/bug_report/venv/share/jupyter/lab
```

Note: I did not install jupyterlab-vim into my local virtual environment, but my  .local/share directory has it anyway.

**Comment by adchrist** [comment]
Just sharing what worked for me, if anyone else comes across this issue while searching. 

I went into the jupyter lab extensions folder at: 

`cd ~/.local/share/jupyter/labextensions/@jupyterlab`

and removed the 'git' folder. Then I used pip to uninstall jupyterlab-git, to be doubly sure. When I installed it again, through pip, the versions matched and it works again. Hope this helps someone else!
